### ISSUE TITLE
make EMULATOR=sims fail

### ISSUE THREAD

**Issue by TheFausap** [PDP-10/its#874]
Hello,
I'm trying to build the system with "make EMULATOR=sims", but after a while the build got a timeout with this error:

*:load sysbin;xxfile bin
*ttyop1$b   $g
$^K
*:PDUMP SYS2;TS XXFILE
DB     HACTRN SYS2   _DUMP_ OUTPUT WRITE  16:16:27
DB     HACTRN SYS2   TS     XXFILE DELRNM 16:16:27
*$P
$1B; TTYOP1>>SETZM TTYQ   kill
?U?

I'm using Ubuntu 18.04.

I did the same build under CentOS 6.9 and it worked.

regards,
Fausto

**Comment by larsbrinkhoff** [comment]
XXFILE -> @atsampson.

Are you running Ubuntu and CentOS on different machines?

We have a long-standing bug where build stall for no reason.  But ?U? looks like a different problem.

**Comment by atsampson** [comment]
Looks like the build script typed `:kill\r` after seeing the PDUMP line, but something ate the `:` before DDT got to see it. In my build log it says:

```
$1B; TTYOP1>>SETZM TTYQ   :kill
```

Maybe it should wait for `>>` - or we could invoke XXFILE in a way that'll let it run to completion - or add an explicit purification entrypoint to the code?

**Comment by TheFausap** [comment]
Yes. I'm running Ubuntu and CentOS on different machines.

**Comment by TheFausap** [comment]
An update: with EMULATOR=simh, under Ubuntu, I have no issue building the system.
So maybe this is a SIMS emulator issue.

**Comment by larsbrinkhoff** [comment]
Thanks for the info.  Various combinations of computers and emulators can exhibit the problem.  We don't know what it is.  I have a vague sense that faster host computers are more likely to fail.  And klh10 is more prone to hang.

**Comment by atsampson** [comment]
It builds OK for me with EMULATOR=sims. Looking at the XXFILE code, it hasn't opened the TTY yet by the time it hits the breakpoint, so I don't think this is a problem with XXFILE as such.

**Comment by TheFausap** [comment]
Another update: I tested with another compiler version, gcc 5.5.0, but the result is always the same.

**Comment by TheFausap** [comment]
Tried the new branch... I got always a timeout but in a different place

```
(ASUM SUPRV COMM DSKFN MLISP MEDIT DB DISPLA GRAM GRIND COMPAR NFORMA TRIGI CSI!
MP MAT MATRIX FACTOR RAT3A RAT3B RAT3C RAT3D RAT3E NRAT4 LESFAC SPGCD SIMP INMI!
S OPERS MLOAD MERROR MFORMT ERMSGM RUNTIM UTILS MUTILS) 
(mapcan #'(lambda (x) (cond ((not (memq x
'(SETS TRANSS MTREE TRHOOK EDLM)
)) (doit x)))) (append todo todoi))
; Unfasl file ((DSK MUNFAS) ASUM UNFASL) not found.
; Unfasl file ((DSK MUNFAS) SERIES UNFASL) not found.
; Unfasl file ((DSK MUNFAS) NUMTH UNFASL) not found.
; Unfasl file ((DSK MUNFAS) SCHATC UNFASL) not found.
; Unfasl file ((DSK MUNFAS) TRGRED UNFASL) not found.
; Unfasl file ((DSK MAXOUT) COMBIN UNFASL) not found.
; Unfasl file ((DSK MUNFAS) SUPRV UNFASL) not found.
; Unfasl file ((DSK MUNFAS) COMM UNFASL) not found.
; Unfasl file ((DSK MAXOUT) COMM2 UNFASL) not found.
; Unfasl file ((DSK MUNFAS) DSKFN UNFASL) not found.
; Unfasl file ((DSK MUNFAS) MLISP UNFASL) not found.

The last command timed out.
Makefile:49: recipe for target 'out/sims/rp03.2' failed
make: *** [out/sims/rp03.2] Error 1
```
This seems a real timeout, maybe we should use a bigger global timeout.

**Comment by larsbrinkhoff** [comment]
Looks like you got past the failure point from before, so this is better.

However, the new problem looks like the one we had before.  I'm sorry, but we don't have a solution for that.  The only workaround is to try EMULATOR=simh or klh10.

**Comment by TheFausap** [comment]
After the previous failure, I did another test: I changed to the newer gcc (7.3.0) and modified, in build/build.tcl, the timeout from 100 to 1000, and the build succeded.

**Comment by larsbrinkhoff** [comment]
Thank you @TheFausap, that's very interesting.  Would you like to repeat the experiment twice, and only change one variable each time?  I.e. one run with the original version of gcc but timeout 1000.  And another run with gcc 7.3.0 and timeout 100.

**Comment by larsbrinkhoff** [comment]
Or maybe gcc 7.3.0 was the version you used initially?

In that case, I think we'd need to take a look at the timeout used at this particular point building Macsyma.

CC @eswenson1

**Comment by eswenson1** [comment]
I will give it a try on jenkins-its and report on the results.  I'm currently using this version of gcc:

> $ gcc --version
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.9) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

**Comment by eswenson1** [comment]
I installed gcc-7 on jenkins-its.  I'm running a test build with the KLH10 emulator (which used to fail all the time). I haven't updated any timeout values yet.

**Comment by eswenson1** [comment]
Well, no dice.  I got:

> *:midas sys1;ts dir_bawden;dir^k
DB     MIDAS  SYS1   _MIDAS OUTPUT DELRNM 21:25:53
DB     MIDAS  SYS1   _MIDAS OUTPUT WRITE  21:25:53
DIR
DIR
Constants area inclusive
From    To
22327   22410
Run
DB     MIDAS  SYS1   TS     DIR    DELRNM 21:25:53
time = 0.00
1475 Symbols including initial ones (54% used)

:KILL
*
The last command timed out.
Makefile:46: recipe for target 'out/klh10/rp0.dsk' failed
make: *** [out/klh10/rp0.dsk] Error 1
jenkins@localhost:~/workspace/pdp-10-its-klh10$

**Comment by TheFausap** [comment]
Update: reverting to 100, makes the build failing again.
So to recap, using the new branch (ats/bug874)

1) build.tcl orig, gcc 5.5.0 : KO
2) build.tcl new, gcc 7.3.0 : OK
3) build.tcl orig, gcc 7.3.0 : KO

I'm trying to build again using build.tcl new and gcc 5.5.0, I did it with these settings only with the old branch.

**Comment by larsbrinkhoff** [comment]
Eric, I think your failure is the mystery problem we discussed before.

@TheFausap initially couldn't get past building XXFILE, seemingly due to a problem in Expect.  Now there's a timeout building Macsyma, but I think it's just a plain slow build step that times out.

**Comment by larsbrinkhoff** [comment]
Fausto, can you try the lars/macsyma branch?  It sets a longer timeout in the build script which I think is responsible for your problem.

**Comment by TheFausap** [comment]
Hello Lars, that branch works without any problem and any mods, under Ubuntu 18.04 and gcc 7.3.0

**Comment by larsbrinkhoff** [comment]
Thanks for checking!  We'll have the fix merged soon.

**Comment by TheFausap** [comment]
Update: I also completed my last check and I can confirm that the branch ats/bug874 works well also with gcc 5.5.0, so 

4. build.tcl new, gcc 5.5.0 : OK

**Comment by larsbrinkhoff** [comment]
Hello again,

It would be good if you had time to check the new fix in #896.  Thanks.
